testeng/jobs/upgradePythonRequirements.groovy

```diff
@@ -0,0 +1,66 @@
+package testeng
+
+Map edxPlatform = [
+    repoName: 'edx-platform',
+]
+
+List jobConfigs = [
+    edxPlatform
+]
+
+/* Iterate over the job configurations */
+jobConfigs.each { jobConfig ->
+
+    job("${jobConfig.repoName}-upgrade-python-requirements") {
+
+        logRotator {
+            daysToKeep(14)
+        }
+        concurrentBuild(false)
+        label('jenkins-worker')
+        environmentVariables(
+            REPO_NAME: "${jobConfig.repoName}"
+        )
+        multiscm {
+            git {
+                remote {
+                    url("https://github.com/edx/${jobConfig.repoName}.git")
+                }
+                branch('master')
+                extensions {
+                    cleanBeforeCheckout()
+                    relativeTargetDirectory("${jobConfig.repoName}")
+                }
+            }
+            git {
+                remote {
+                    url('https://github.com/edx/testeng-ci.git')
+                }
+                branch('youngstrom/update-github-scripts')
+                extensions {
+                    cleanBeforeCheckout()
+                    relativeTargetDirectory('testeng-ci')
+                }
+            }
+        }
+        triggers {
+            cron("@daily")
```

Comment:
Can the frequency be added as a parameter in `jobConfigs`?  We'll definitely want this to vary across repositories, depending on how often we modify them.